### ISSUE TITLE
Configure SMTP over OAuth to send emails

### ISSUE THREAD

**Issue by corrad82-4s** [DSpace/DSpace#8627]
**Is your feature request related to a problem? Please describe.**
Google, but maybe soon also other like Microsoft, email providers requires OAuth to establish and authenticate on a SMTP connection. DSpace should support this flow.

**Describe the solution you'd like**
The user has the possibility of setting up the email connection via settings, then ideal flow behaves as follows: 
* User chooses SMTP over OAUTH2 as their option and provides the email address to be used to authenticate.
* DSpace opens up a web browser and sends the user to the email provider to sign in
* After successful sign in, the email provider returns an authentication token
* DSpace stores the authentication token for use in the SMTP connections. 

**Additional context**
This task is not trivial and carries some complexity, for example we need to identify a safe place where token to be used in SMTP connections can be stored (a metadata? A file? other?) and also a scheduled process is needed to keep the token up to date.

**Comment by tdonohue** [comment]
I agree we should support this.  However, I'm not sure it is as complex as you describe.  My understanding (and I admit I could be wrong) is that you could simply "register" your application (DSpace), generate the necessary tokens, and then store them in DSpace's configuration (similar to how we already have SMTP settings in dspace.cfg).   Here's a basic example for Google: https://www.labnol.org/gmail-smtp-send-emails-220530 (and I think you can do something similar for Microsoft).

So, I might be wrong, but I don't know that we need to have the step where DSpace has you login & then stores the token for later.  We may be able to simply add some new configurations specific to SMTP over OAuth and provide instructions on how to register DSpace as an OAuth Client, and then generate the necessary tokens to save in DSpace's configuration.

In any case, I agree this should be supported... just noting that we may want to see if there's an easier route here than the "non-trivial" solution you described above.

**Comment by abhishekd2t** [comment]
@tdonohue - Rightly mentioned by you, we can have all the required parameters as properties under the configuration file and use it when it is needed.

**Comment by tdonohue** [comment]
Thanks @abhishekd2t .  As you requested via email, I'm assigning this to you for the DSquare team to work on.

**Comment by aseyedia** [comment]
Just voicing some demand for this features, apparently Google began enforcing OAuth authentication so our email server no longer works.

**Comment by tdonohue** [comment]
It looks like this hasn't had any activity, so I'm putting it up for any volunteers to work on again.  Needs a volunteer

**Comment by aseyedia** [comment]
@tdonohue We are interested in potentially developing this feature since our workaround might be a dead-end. Could you nudge us in the right direction on where to get started? I'm also looking at the issues myself as well.

**Comment by tdonohue** [comment]
@aseyedia : I'm afraid I don't have any great advice on this one.  I'm not as familiar myself with SMTP over OAuth or how it might be implemented in a Java application. The only advice I had to offer is what I said above: https://github.com/DSpace/DSpace/issues/8627#issuecomment-1442500099

**Comment by mwoodiupui** [comment]
@aseyedia I would start reading here:  https://javaee.github.io/javamail/OAuth2
Or ask your favorite search engine for "java smtp oauth2".

DSpace's email Session is built in `dspace-services/src/main/java/org/dspace/services/email/EmailServiceImpl.java`.

The examples I find are obsessed with *receiving* mail and don't talk much about sending, but I think that the property `mail.smtp.authentication.mechanisms` should be known and a sufficiently new version of Jakarta Mail should understand the value "XOAUTH2".  (I have *not* tried this.)  https://jakarta.ee/specifications/mail/2.0/jakarta-mail-spec-2.0#protocol-selection-and-defaults

**Comment by aseyedia** [comment]
Just a heads up – I think I can only do this for Google for now; that is where our particular need is and I don't have an MS Client ID and Secret to experiment with, though I might be able to add this down the road.
